src/ir/context.rs

```diff
@@ -2430,17 +2437,27 @@ If you encounter an error missing from this list, please file an issue or a PR!"
         self.allowlisted = Some(allowlisted);
         self.codegen_items = Some(codegen_items);
 
+        let mut warnings = Vec::new();
+
         for item in self.options().allowlisted_functions.unmatched_items() {
-            warn!("unused option: --allowlist-function {}", item);
+            let warn = format!("unused option: --allowlist-function {}", item);
+            warn!("{}", warn);
+            warnings.push(warn);
         }
 
         for item in self.options().allowlisted_vars.unmatched_items() {
-            warn!("unused option: --allowlist-var {}", item);
+            let warn = format!("unused option: --allowlist-var {}", item);
+            warn!("{}", warn);
+            warnings.push(warn);
         }
 
         for item in self.options().allowlisted_types.unmatched_items() {
-            warn!("unused option: --allowlist-type {}", item);
+            let warn = format!("unused option: --allowlist-type {}", item);
+            warn!("{}", warn);
+            warnings.push(warn);
```

Comment:
Rather than duplicating all this, what about just changing:

```
warn!("unused option: ...");
```

with:

```
self.warnings.push(format!(...))
```

Then just call `warn!` at the end on a loop?